lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -122,7 +123,12 @@ public void CopyToLocal(Uri remoteFileUri, string localFileName)
             }
             if (status.EntriesFailed.Count != 0)
             {
-                throw new IOException($"{status.EntriesFailed.Count} entries did not get transferred correctly");
+                var failedEntriesMsg = new StringBuilder();
+                foreach (var entry in status.EntriesFailed)
+                {
+                    failedEntriesMsg.Append($"Entry {entry.EntryName} failed with error message {entry.Errors}. ");
```

Comment:
This message might get very large if entries failed is a very large number like 1000+ . Better to clip it in that case. I am guessing if failure happens the error msg for the 1000th entry will be same as all the others